client/web/compose/src/components/Public/Record/Modal.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <b-modal
+    id="record-modal"
+    v-model="showRecordModal"
+    scrollable
+    body-class="p-0"
+    footer-class="p-0"
+    size="xl"
+    @hidden="hideModal"
+  >
+    <template #modal-title>
+      <portal-target
+        name="record-modal-header"
+      />
+    </template>
+
+    <view-record
+      :namespace="namespace"
+      :page="page"
+      :module="module"
+      :record-i-d="recordID"
+      :show-record-modal="showRecordModal"
+    />
+
+    <template #modal-footer>
+      <portal-target
+        name="record-modal-footer"
+        class="w-100 m-0"
+      />
+    </template>
+  </b-modal>
+</template>
+
+<script>
+import record from 'corteza-webapp-compose/src/mixins/record'
+import { compose } from '@cortezaproject/corteza-js'
+import ViewRecord from 'corteza-webapp-compose/src/views/Public/Pages/Records/View'
+import { mapGetters } from 'vuex'
+
+export default {
+  i18nOptions: {
+    namespaces: 'block',
+  },
+
+  name: 'RecordModal',
+
+  components: {
+    ViewRecord,
+  },
+
+  mixins: [
+    record,
+  ],
+
+  props: {
+    namespace: {
+      type: compose.Namespace,
+      required: true,
+    },
+  },
+
+  data () {
+    return {
+      showRecordModal: false,
+      recordID: null,
+      module: null,
+      page: null,
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      getModuleByID: 'module/getByID',
+      pages: 'page/set',
+    }),
+  },
+
+  created () {
+    this.$root.$on('show-record-modal', this.showModal)
+
+    const { modalRecordID, modalModuleID, modalPageID } = this.$route.query
+    if (modalRecordID) {
+      this.showModal({
+        recordID: modalRecordID,
+        moduleID: modalModuleID,
+        recordPageID: modalPageID,
+      })
+    }
+  },
+
+  beforeDestroy () {
+    this.$root.$off('show-record-modal')
+  },
+
+  methods: {
+    showModal (e) {
+      const { recordID, moduleID, recordPageID } = e
+      this.showRecordModal = true
+      this.recordID = recordID
+      this.page = this.pages.find(p => p.pageID === recordPageID)
+      this.module = this.getModuleByID(moduleID)
+
+      // persist the modal in the url
+      this.$router.replace({
+        query: {
+          modalRecordID: recordID,
```

Comment:
Not sure why you resolved my previous comment regarding this, but no need to call the variables with the modal prefix. Just use recordID and so on